cupy/creation/from_data.py

```diff
@@ -41,6 +44,20 @@ def array(obj, dtype=None, copy=True, order='K', subok=False, ndmin=0):
     return core.array(obj, dtype, copy, order, subok, ndmin)
 
 
+def _convert_object_with_cuda_array_interface(a):
+    desc = a.__cuda_array_interface__
+    shape = desc['shape']
+    dtype = numpy.dtype(desc['typestr'])
+    nbytes = numpy.prod(shape) * dtype.itemsize
```

Comment:
Let me know how numpy broadcast array is dealt with by cupy. It looks that the broadcast array is expanded so that there is no overlap when converting array from numpy to cupy. Is this correct?

    >>> x = numpy.broadcast_to(numpy.ones(4), (4,4))
    >>> x.strides
    (0, 8)
    >>> y = cupy.array(x)
    >>> y.strides
    (32, 8)

Reply:
CuPy supports `0` in strides. You can safely use it.

```py
>>> x = cp.broadcast_to(cp.ones(4), (4,4))
>>> x.strides
(0, 8)
>>> x.sum()
array(16.)
```

Reply:
I see... Anyway let's do what we think we're correct.

Reply:
I guess you may have already noted, it's reported in https://github.com/numba/numba/issues/3550.